### PR TITLE
XMLHttpRequest: timeout and multiple fetches

### PR DESCRIPTION
**Pull request by annevk**
See https://www.w3.org/Bugs/Public/show_bug.cgi?id=28117.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5104/)

**Comment by annevk** [comment]
Chrome bug (only browser that fails as far as I can tell): https://bugs.chromium.org/p/chromium/issues/detail?id=699985.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/209359633)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 23d5bc5c03d6c8aa20fac4eeda3bc863564b3407
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>17 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/abort-after-stop.htm">/XMLHttpRequest/abort-after-stop.htm</a></summary>


|                               Subtest                                | Results |                   Messages                   |
|----------------------------------------------------------------------|---------|----------------------------------------------|
|                                                                      | TIMEOUT |                                              |
| `XMLHttpRequest: abort event should fire when stop() method is used` | FAIL    | `assert_equals: expected true but got false` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/abort-during-upload.htm">/XMLHttpRequest/abort-during-upload.htm</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `XMLHttpRequest: abort() while sending data` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/event-timeout-order.htm">/XMLHttpRequest/event-timeout-order.htm</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `XMLHttpRequest: event - timeout (order of events)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/event-timeout.htm">/XMLHttpRequest/event-timeout.htm</a></summary>


|             Subtest             | Results | Messages |
|---------------------------------|---------|----------|
|                                 | OK      |          |
| `XMLHttpRequest: timeout event` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/open-url-redirected-worker-origin.htm">/XMLHttpRequest/open-url-redirected-worker-origin.htm</a></summary>


|                             Subtest                              | Results |                                                                                                               Messages                                                                                                              |
|------------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                  | OK      |                                                                                                                                                                                                                                     |
| `XMLHttpRequest: redirected worker scripts, origin and referrer` | PASS    |                                                                                                                                                                                                                                     |
| `Referer header`                                                 | FAIL    | `assert_equals: expected "referer: http://web-platform.test:8000/XMLHttpRequest/resources/workerxhr-origin-referrer.js\n" but got "Referer: http://web-platform.test:8000/XMLHttpRequest/resources/workerxhr-origin-referrer.js\n"` |
| `Origin header`                                                  | FAIL    | `assert_equals: expected "origin: http://web-platform.test:8000\n" but got "Origin: http://web-platform.test:8000\n"`                                                                                                               |
| `Request URL test`                                               | PASS    |                                                                                                                                                                                                                                     |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/preserve-ua-header-on-redirect.htm">/XMLHttpRequest/preserve-ua-header-on-redirect.htm</a></summary>


|                            Subtest                             | Results | Messages |
|----------------------------------------------------------------|---------|----------|
|                                                                | OK      |          |
| `XMLHttpRequest: User-Agent header is preserved on redirect`   | PASS    |          |
| `XMLHttpRequest: User-Agent header is preserved on redirect 1` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/responseurl.html">/XMLHttpRequest/responseurl.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `404 response has proper responseURL`        | PASS    |          |
| `Redirected response has proper responseURL` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-bogus-sync.htm">/XMLHttpRequest/send-redirect-bogus-sync.htm</a></summary>


|                                               Subtest                                                | Results | Messages |
|------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                      | OK      |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (301: foobar://abcd)`              | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (302: http://z)`                   | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (302: mailto:someone@example.org)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (303: http://z)`                   | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (303: tel:1234567890)`             | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-bogus.htm">/XMLHttpRequest/send-redirect-bogus.htm</a></summary>


|                                            Subtest                                             | Results | Messages |
|------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                | OK      |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (302: http://example.not)`         | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (302: mailto:someone@example.org)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (303: http://example.not)`         | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (303: foobar:someone@example.org)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-post-upload.htm">/XMLHttpRequest/send-redirect-post-upload.htm</a></summary>


|                                 Subtest                                  | Results | Messages |
|--------------------------------------------------------------------------|---------|----------|
|                                                                          | OK      |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (301)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (302)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (303)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-to-cors.htm">/XMLHttpRequest/send-redirect-to-cors.htm</a></summary>


|                                         Subtest                                          | Results | Messages |
|------------------------------------------------------------------------------------------|---------|----------|
|                                                                                          | OK      |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (301)`                       | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (302)`                       | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (303)`                       | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307)`                       | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with null)`        | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with string)`      | PASS    |          |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with typed array)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-to-non-cors.htm">/XMLHttpRequest/send-redirect-to-non-cors.htm</a></summary>


|                                       Subtest                                        | Results | Messages |
|--------------------------------------------------------------------------------------|---------|----------|
|                                                                                      | OK      |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (301)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (302)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (303)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect.htm">/XMLHttpRequest/send-redirect.htm</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `XMLHttpRequest: send() - Redirects (basics) (301)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (302)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (303)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-sync-blocks-async.htm">/XMLHttpRequest/send-sync-blocks-async.htm</a></summary>


|                                    Subtest                                    | Results |      Messages      |
|-------------------------------------------------------------------------------|---------|--------------------|
|                                                                               | OK      |                    |
| `XMLHttpRequest: sync requests should block events on pending async requests` | FAIL    | `t is not defined` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/setrequestheader-after-send.htm">/XMLHttpRequest/setrequestheader-after-send.htm</a></summary>


|                      Subtest                      | Results | Messages |
|---------------------------------------------------|---------|----------|
|                                                   | OK      |          |
| `XMLHttpRequest: setRequestHeader() after send()` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/template-element.html">/XMLHttpRequest/template-element.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `XMLHttpRequest: template element parsing`   | PASS    |          |
| `XMLHttpRequest: template element parsing 1` | PASS    |          |
| `XMLHttpRequest: template element parsing 2` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/timeout-multiple-fetches.html">/XMLHttpRequest/timeout-multiple-fetches.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `Redirects should not reset the timer`       | PASS    |          |
| `CORS preflights should not reset the timer` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/209359636)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 23d5bc5c03d6c8aa20fac4eeda3bc863564b3407
Using browser at version 58.0.3029.6 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>17 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/abort-after-stop.htm">/XMLHttpRequest/abort-after-stop.htm</a></summary>


|                               Subtest                                | Results | Messages |
|----------------------------------------------------------------------|---------|----------|
|                                                                      | TIMEOUT |          |
| `XMLHttpRequest: abort event should fire when stop() method is used` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/abort-during-upload.htm">/XMLHttpRequest/abort-during-upload.htm</a></summary>


|                   Subtest                    | Results |                                             Messages                                            |
|----------------------------------------------|---------|-------------------------------------------------------------------------------------------------|
|                                              | OK      |                                                                                                 |
| `XMLHttpRequest: abort() while sending data` | FAIL    | `assert_equals: expected "upload.loadstart(0,9999,true)" but got "upload.loadstart(0,0,false)"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/event-timeout-order.htm">/XMLHttpRequest/event-timeout-order.htm</a></summary>


|                       Subtest                       | Results |                                            Messages                                           |
|-----------------------------------------------------|---------|-----------------------------------------------------------------------------------------------|
|                                                     | OK      |                                                                                               |
| `XMLHttpRequest: event - timeout (order of events)` | FAIL    | `assert_equals: expected "upload.loadstart(0,12,true)" but got "upload.loadstart(0,0,false)"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/event-timeout.htm">/XMLHttpRequest/event-timeout.htm</a></summary>


|             Subtest             | Results | Messages |
|---------------------------------|---------|----------|
|                                 | OK      |          |
| `XMLHttpRequest: timeout event` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/open-url-redirected-worker-origin.htm">/XMLHttpRequest/open-url-redirected-worker-origin.htm</a></summary>


|                             Subtest                              | Results |                                                                                                                         Messages                                                                                                                         |
|------------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                  | OK      |                                                                                                                                                                                                                                                          |
| `XMLHttpRequest: redirected worker scripts, origin and referrer` | PASS    |                                                                                                                                                                                                                                                          |
| `Referer header`                                                 | FAIL    | `assert_equals: expected "referer: http://web-platform.test:8000/XMLHttpRequest/resources/workerxhr-origin-referrer.js\n" but got "Referer: http://web-platform.test:8000/XMLHttpRequest/resources/redirect.py?location=workerxhr-origin-referrer.js\n"` |
| `Origin header`                                                  | FAIL    | `assert_equals: expected "origin: http://web-platform.test:8000\n" but got "Origin: http://web-platform.test:8000\n"`                                                                                                                                    |
| `Request URL test`                                               | PASS    |                                                                                                                                                                                                                                                          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/preserve-ua-header-on-redirect.htm">/XMLHttpRequest/preserve-ua-header-on-redirect.htm</a></summary>


|                            Subtest                             | Results |                                                                                    Messages                                                                                    |
|----------------------------------------------------------------|---------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                | OK      |                                                                                                                                                                                |
| `XMLHttpRequest: User-Agent header is preserved on redirect`   | PASS    |                                                                                                                                                                                |
| `XMLHttpRequest: User-Agent header is preserved on redirect 1` | FAIL    | `assert_equals: expected "User-Agent: TEST\n" but got "User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.6 Safari/537.36\n"` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/responseurl.html">/XMLHttpRequest/responseurl.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `404 response has proper responseURL`        | PASS    |          |
| `Redirected response has proper responseURL` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-bogus-sync.htm">/XMLHttpRequest/send-redirect-bogus-sync.htm</a></summary>


|                                               Subtest                                                | Results | Messages |
|------------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                      | OK      |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (301: foobar://abcd)`              | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (302: http://z)`                   | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (302: mailto:someone@example.org)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (303: http://z)`                   | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header; sync) (303: tel:1234567890)`             | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-bogus.htm">/XMLHttpRequest/send-redirect-bogus.htm</a></summary>


|                                            Subtest                                             | Results | Messages |
|------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                | OK      |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (302: http://example.not)`         | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (302: mailto:someone@example.org)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (303: http://example.not)`         | PASS    |          |
| `XMLHttpRequest: send() - Redirects (bogus Location header) (303: foobar:someone@example.org)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-post-upload.htm">/XMLHttpRequest/send-redirect-post-upload.htm</a></summary>


|                                 Subtest                                  | Results | Messages |
|--------------------------------------------------------------------------|---------|----------|
|                                                                          | OK      |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (301)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (302)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (303)` | PASS    |          |
| `XMLHttpRequest: The send() method: POSTing to URL that redirects (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-to-cors.htm">/XMLHttpRequest/send-redirect-to-cors.htm</a></summary>


|                                         Subtest                                          | Results |                 Messages                |
|------------------------------------------------------------------------------------------|---------|-----------------------------------------|
|                                                                                          | OK      |                                         |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (301)`                       | FAIL    | `assert_equals: expected 200 but got 0` |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (302)`                       | FAIL    | `assert_equals: expected 200 but got 0` |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (303)`                       | FAIL    | `assert_equals: expected 200 but got 0` |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307)`                       | FAIL    | `assert_equals: expected 200 but got 0` |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with null)`        | PASS    |                                         |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with string)`      | PASS    |                                         |
| `XMLHttpRequest: send() - Redirect to CORS-enabled resource (307 post with typed array)` | PASS    |                                         |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect-to-non-cors.htm">/XMLHttpRequest/send-redirect-to-non-cors.htm</a></summary>


|                                       Subtest                                        | Results | Messages |
|--------------------------------------------------------------------------------------|---------|----------|
|                                                                                      | OK      |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (301)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (302)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (303)` | PASS    |          |
| `XMLHttpRequest: send() - Redirect to cross-origin resource, not CORS-enabled (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-redirect.htm">/XMLHttpRequest/send-redirect.htm</a></summary>


|                       Subtest                       | Results | Messages |
|-----------------------------------------------------|---------|----------|
|                                                     | OK      |          |
| `XMLHttpRequest: send() - Redirects (basics) (301)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (302)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (303)` | PASS    |          |
| `XMLHttpRequest: send() - Redirects (basics) (307)` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/send-sync-blocks-async.htm">/XMLHttpRequest/send-sync-blocks-async.htm</a></summary>


|                                    Subtest                                    | Results |      Messages      |
|-------------------------------------------------------------------------------|---------|--------------------|
|                                                                               | OK      |                    |
| `XMLHttpRequest: sync requests should block events on pending async requests` | FAIL    | `t is not defined` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/setrequestheader-after-send.htm">/XMLHttpRequest/setrequestheader-after-send.htm</a></summary>


|                      Subtest                      | Results | Messages |
|---------------------------------------------------|---------|----------|
|                                                   | OK      |          |
| `XMLHttpRequest: setRequestHeader() after send()` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/template-element.html">/XMLHttpRequest/template-element.html</a></summary>


|                   Subtest                    | Results | Messages |
|----------------------------------------------|---------|----------|
|                                              | OK      |          |
| `XMLHttpRequest: template element parsing`   | PASS    |          |
| `XMLHttpRequest: template element parsing 1` | PASS    |          |
| `XMLHttpRequest: template element parsing 2` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5104/XMLHttpRequest/timeout-multiple-fetches.html">/XMLHttpRequest/timeout-multiple-fetches.html</a></summary>


|                   Subtest                    | Results |                          Messages                         |
|----------------------------------------------|---------|-----------------------------------------------------------|
|                                              | OK      |                                                           |
| `Redirects should not reset the timer`       | PASS    |                                                           |
| `CORS preflights should not reset the timer` | FAIL    | `assert_unreached: Not possible Reached unreachable code` |

</details>
</details>

File: XMLHttpRequest/resources/redirect.py [hunk 1]
```diff
@@ -1,8 +1,14 @@
+import time
+
 def main(request, response):
     code = int(request.GET.first("code", 302))
     location = request.GET.first("location", request.url_parts.path +"?followed")
 
-    if request.url.endswith("?followed"):
+    if "delay" in request.GET:
+        delay = float(request.GET.first("delay", 500))
```
Comment (foolip) [inline XMLHttpRequest/resources/redirect.py]: Is the fallback 500 ever used given the if condition we're in?

**Review by foolip — APPROVED** [review]
Fixed my own pet peeve, LGTM with or without a fix for the fallback 500 thing, if it's legit.

File: XMLHttpRequest/resources/redirect.py (hunk 1 above)
Comment (annevk) [inline XMLHttpRequest/resources/redirect.py]: It's not I think. Just copypasta. Remove it?

Comment (foolip) [inline XMLHttpRequest/resources/redirect.py]: Yep, see what happens. On a phone now, so over and out.

Comment (annevk) [inline XMLHttpRequest/resources/redirect.py]: Same results, so will land once checks pass.

**Pull request merged**